core/src/main/java/com/linecorp/armeria/internal/common/RequestContextThreadLocalAccessor.java

```diff
@@ -0,0 +1,132 @@
+import io.micrometer.context.ContextSnapshot.Scope;
+import io.micrometer.context.ThreadLocalAccessor;
+
+/**
+ * This class works with the
+ * <a href="https://docs.micrometer.io/context-propagation/reference/index.html">Micrometer
+ * Context Propagation</a> to keep the {@link RequestContext} during
+ * <a href="https://github.com/reactor/reactor-core">Reactor</a> operations.
+ * Get the {@link RequestContextThreadLocalAccessor} to register it to the {@link ContextRegistry}.
+ * Then, {@link ContextRegistry} will use {@link RequestContextThreadLocalAccessor} to
+ * propagate context during the
+ * <a href="https://github.com/reactor/reactor-core">Reactor</a> operations
+ * so that you can get the context using {@link RequestContext#current()}.
+ * However, please note that you should include Mono#contextWrite(ContextView) or
+ * Flux#contextWrite(ContextView) to end of the Reactor codes.
+ * If not, {@link RequestContext} will not be keep during Reactor Operation.
+ */
+public final class RequestContextThreadLocalAccessor implements ThreadLocalAccessor<RequestContext> {
+
+    private static final Object KEY = RequestContext.class;
+
+    /**
+     * The value which obtained through {@link RequestContextThreadLocalAccessor},
+     * will be stored in the Context under this {@code KEY}.
+     * This method will be called by {@link ContextSnapshot} internally.
+     */
+    @Override
+    public Object key() {
+        return KEY;
+    }
+
+    /**
+     * {@link ContextSnapshot} will call this method during the execution
+     * of lambda functions in {@link ContextSnapshot#wrap(Runnable)},
+     * as well as during Mono#subscribe(), Flux#subscribe(),
+     * {@link Subscription#request(long)}, and CoreSubscriber#onSubscribe(Subscription).
+     * Following these calls, {@link ContextSnapshot#setThreadLocals()} is
+     * invoked to restore the state of {@link RequestContextStorage}.
+     * Furthermore, at the end of these methods, {@link Scope#close()} is executed
+     * to revert the {@link RequestContextStorage} to its original state.
+     */
+    @Override
+    public RequestContext getValue() {
+        return RequestContext.currentOrNull();
+    }
+
+    /**
+     * {@link ContextSnapshot} will call this method during the execution
+     * of lambda functions in {@link ContextSnapshot#wrap(Runnable)},
+     * as well as during Mono#subscribe(), Flux#subscribe(),
+     * {@link Subscription#request(long)}, and CoreSubscriber#onSubscribe(Subscription).
+     * Following these calls, {@link ContextSnapshot#setThreadLocals()} is
+     * invoked to restore the state of {@link RequestContextStorage}.
+     * Furthermore, at the end of these methods, {@link Scope#close()} is executed
+     * to revert the {@link RequestContextStorage} to its original state.
+     */
+    @Override
+    @SuppressWarnings("MustBeClosedChecker")
+    public void setValue(RequestContext value) {
+        value.push();
```

Comment:
Let's just use `RequestContextUtil.getAndSet(value);`.
`value.push()` internally invokes a hook but the `closeable` never be closed because we discard the result of `value.push()`.
https://github.com/line/armeria/blob/87f231848f5eb8b0d25061204de2a34a9fa37b73/core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java#L192
https://github.com/line/armeria/blob/87f231848f5eb8b0d25061204de2a34a9fa37b73/core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java#L197

This is a limitation of the current design of Mircometer context-proparation.
Because we cannot call the `closeable`, we should avoid invoking the hook altogether and inform users of the limitation of this API.

Reply:
@minwoox nim, thanks for your comments. 
`RequestContextUtil.getAndSet(value)` seems not get `SafeClosable` instance. 
I make new commit to apply your comments. 
Please take another look, when you have free time 🙇‍♂️

Reply:
Still looks good. Thanks!